In [1]:
# Necessary Libraries

import numpy as np, pandas as pd

In [2]:
# Lets import the data

data = pd.read_excel(r'C:\Users\joshu\Desktop\prwa Tech\Assignments\3 Movies\Workings\youtube(2).xlsx', header= None)

In [3]:
# Lets inspect the data

data.head()

0                     1       2               3      4       5   \
0  QuRYeRnAuXM  EvilSquirrelPictures  1135.0  Pets & Animals  252.0  1075.0   
1  3TYqkBJ9YRk                hggh22  1135.0          Comedy  169.0   228.0   
2  rSJ8QZWBegU               TimeGem  1135.0   Entertainment   95.0   356.0   
3  nRcovJn9xHg            wooochacha  1135.0   Entertainment  118.0  1115.0   
4  UEvVksP91kg              johnx113  1135.0   Entertainment   83.0   281.0   

     6     7     8            9   ...           19           20           21  \
0  4.96  46.0  86.0  gFa1YMEJFag  ...  8qQrrfUTmh0  zQ83d_D2MGs  u6_DQQjLsAw   
1  5.00   5.0   3.0  QuRYeRnAuXM  ...  g9e1alirMhc  YZev1imoxX8  I4yKEK9o8gA   
2  4.31  13.0   1.0  QuRYeRnAuXM  ...  ShhClb6J-NA  g9e1alirMhc  YZev1imoxX8   
3  2.23  57.0  73.0  QuRYeRnAuXM  ...  YZev1imoxX8  I4yKEK9o8gA  zQ83d_D2MGs   
4  2.67   9.0  16.0  gFa1YMEJFag  ...  g9e1alirMhc  YZev1imoxX8  I4yKEK9o8gA   

            22           23           24           25           26  \
0  73Wz9CQFDtE          NaN          NaN          NaN          NaN   
1  zQ83d_D2MGs  1GKaVzNDbuI  yuZhwV24PmM  DomumdGQSg8  hiSmlmXp-aU   
2  I4yKEK9o8gA  zQ83d_D2MGs  1GKaVzNDbuI  yuZhwV24PmM  DomumdGQSg8   
3  1GKaVzNDbuI  yuZhwV24PmM  DomumdGQSg8  hiSmlmXp-aU  pFUYi7dp1WU   
4  zQ83d_D2MGs  1GKaVzNDbuI  yuZhwV24PmM  DomumdGQSg8  hiSmlmXp-aU   

            27           28  
0          NaN          NaN  
1  pFUYi7dp1WU  2l6vwAIAqNU  
2  hiSmlmXp-aU  pFUYi7dp1WU  
3  2l6vwAIAqNU  WWqed9u6rr4  
4  pFUYi7dp1WU  2l6vwAIAqNU  

[5 rows x 29 columns]

In [4]:
# Lets drop the columns which are less usefull

data = data.iloc[:,:7]

In [5]:
# Lets impute the Column name

data.columns =['Id','Admin','Interval','Category','Length','Views','Rating']

In [6]:
data

Id                 Admin  Interval        Category  Length  \
0     QuRYeRnAuXM  EvilSquirrelPictures    1135.0  Pets & Animals   252.0   
1     3TYqkBJ9YRk                hggh22    1135.0          Comedy   169.0   
2     rSJ8QZWBegU               TimeGem    1135.0   Entertainment    95.0   
3     nRcovJn9xHg            wooochacha    1135.0   Entertainment   118.0   
4     UEvVksP91kg              johnx113    1135.0   Entertainment    83.0   
...           ...                   ...       ...             ...     ...   
4095  esCvu46qDaA                 RSMFD     863.0   Entertainment    63.0   
4096  kOZ8bRAO7YQ         RockstarGames       0.0            UNA     63.0   
4097  GSsVZcKHues               GTASite    1024.0   Entertainment   123.0   
4098  foKRqJGcbPQ       scrambledeggsTV    1100.0   Entertainment   136.0   
4099          NaN               rehab07     771.0   Entertainment   186.0   

         Views  Rating  
0       1075.0    4.96  
1        228.0    5.00  
2        356.0    4.31  
3       1115.0    2.23  
4        281.0    2.67  
...        ...     ...  
4095  116929.0    4.84  
4096  199361.0    4.79  
4097  486752.0    4.86  
4098   13115.0    4.45  
4099  103556.0    4.30  

[4100 rows x 7 columns]

### 1. Top 5 categories with Max no. of videos uploaded

In [7]:
t5_cat = pd.DataFrame(data.Category.value_counts().reset_index())
t5_cat.columns = ['video_category', 'video_counts']
t5_cat.iloc[:5,:]

video_category  video_counts
0    Entertainment           908
1            Music           862
2           Comedy           414
3   People & Blogs           398
4  News & Politics           333

### 2. Top 10 Rated videos

In [11]:
ratings_5 = data[data.Rating == 5.0]

In [32]:
ratings_5.shape

(755, 7)

##### As we have 755 videos with 5.0 as ratings....lets order with no. of views

In [18]:
ratings_5 = ratings_5.sort_values('Views', ascending= False)

In [31]:
ratings_5.iloc[:10,:].loc[:,['Id','Category', 'Rating', 'Views']]

Id          Category  Rating     Views
2340  nUDksaQp3IA     Entertainment     5.0  101016.0
3591  K4i7PEClSF8     Entertainment     5.0   75573.0
2436  nwqXVAYsYW4             Music     5.0   28966.0
2569  hnA0VyR3HI4     Entertainment     5.0   22921.0
3919  od1PLcqUhfs             Music     5.0   22118.0
2863  Yx2iza6EfZo             Music     5.0   21049.0
2286  8o72MzrgFiU  Autos & Vehicles     5.0   17824.0
2044  gjbG5N4EfDc             Music     5.0   14752.0
1011  kXeamc7e31c     Howto & Style     5.0   14413.0
2410  MIQFTBsGccA  Film & Animation     5.0   13103.0

### 3. Most viewed videos

In [29]:
view_sorted = data.sort_values('Views', ascending= False)
view_sorted.iloc[0,:]

Id               12Z3J1uzd0Q
Admin                kaejane
Interval               404.0
Category    Film & Animation
Length                 615.0
Views             65341925.0
Rating                  3.03
Name: 1109, dtype: object

#### ID - 12Z3J1uzd0Q
#### Category - Film & Animation
#### Views - 6,53,41,925